### DISCUSSION TITLE
MIDI computer shutdown

### DISCUSSION THREAD

**Discussion by ppytprs** [GrandOrgue/grandorgue#1633] [Help]
Hi. I'm building an organ where the computer will be inaccessible inside the case. I need to be able to shut it down ideally with a MIDI button. Is there a way to do that? I found a patch that someone had made on a different discussion site, assuming an old one. But I can't figure out how to apply a patch. 

Thanks!

**Comment by jason-barba** [comment]
What operating system are you planning to use?

**Comment by ppytprs** [comment]
Windows 10

**Comment by davidgritter** [comment]
I've done this several ways, but using a midi button might be more difficult in windows than in linux for case 2 below
1. If your computer is a standard atx motherboard, its easy to wire a remote pushbutton switch in parallel with the computers front panel power pushbutton.  This can be used for power up/down or for entering and exiting sleep mode.  If you also need a remote way to power on the computer this is convenient
2. In linux a utility called mididings can intercept midi messages and respond by running a shell script to start a program, or in your case issue the shutdown command.  Its python, but it seems to rely on alsa and jack for midi interface, so may be difficult to port to windows. (this approach is the only one that doesn't require extra hardware)
3. A teensy arduino computer can be programmed to emulate a keyboard over USB.  You could tie a pushbutton to the teensy, and it would then send keystrokes to your computer.  The keystrokes definined "way 4" at https://www.isumsoft.com/windows-10/how-to-shut-down-or-restart-computer-with-windows-10.html should work;  or your startup script in the computer could ensure that a windows command line interface retains focus after everything starts up.   The Teensy can also implement a hardware midi interface that you could connect to whatever hardware has the power down button already on it.  (a raspberry pi pico could also do this at lower cost, but the software to emulate the keyboard is much more difficult to write or understand.)
4. Teensy  or pico can also act as a USB midi device.  You could create a GrandOrgue button object that sends a midi message to the teensy or pico, which would interpret the message and activate a relay in parallel with the computer's front panel power pushbutton.

**Reply by ppytprs** [reply]
Thanks for the suggestions there. I would normally have done number 1, but annoyingly the tiny computer I'm using on this project doesn't have the ability to tap in to the power switch.
